samtranslator/model/api/http_api_generator.py

```diff
@@ -218,7 +218,32 @@ def _construct_api_domain(self, http_api):
             basepaths = self.domain.get("BasePath")
         else:
             basepaths = None
+        basepath_resource_list = self._construct_basepath_mappings(basepaths, http_api)
 
+        # Create the Route53 RecordSetGroup resource
+        record_set_group = None
+        if self.domain.get("Route53") is not None:
+            route53 = self.domain.get("Route53")
+            if route53.get("HostedZoneId") is None and route53.get("HostedZoneName") is None:
+                raise InvalidResourceException(
+                    self.logical_id,
+                    "HostedZoneId or HostedZoneName is required to enable Route53 support on Custom Domains.",
+                )
+            logical_id = logical_id_generator.LogicalIdGenerator(
+                "", route53.get("HostedZoneId") or route53.get("HostedZoneName")
+            ).gen()
+            record_set_group = Route53RecordSetGroup(
+                "RecordSetGroup" + logical_id, attributes=self.passthrough_resource_attributes
+            )
+            if "HostedZoneId" in route53:
+                record_set_group.HostedZoneId = route53.get("HostedZoneId")
+            if "HostedZoneName" in route53 and "HostedZoneId" not in route53:
```

Comment:
`if "HostedZoneName" in route53 and "HostedZoneId" not in route53:` -> `elif "HostedZoneName" in route53:`